### PR TITLE
vcflib 1.0.1

### PR DESCRIPTION
**Pull request by tseemann**
Bioconda requires reviews prior to merging pull-requests (PRs). To facilitate this, once your PR is passing tests and ready to be merged, please add the `please review & merge` label so other members of the bioconda community can have a look at your PR and either make suggestions or merge it. Note that if you are not already a member of the bioconda project (meaning that you can't add this label), please ping `@bioconda/core` so that your PR can be reviewed and merged (please note if you'd like to be added to the bioconda project). Please see https://github.com/bioconda/bioconda-recipes/issues/15332 for more details.

* [ ] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/contributor/guidelines.html).
* [ ] This PR adds a new recipe.
* [ ] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by tseemann** [comment]
```
00:33:01 BIOCONDA INFO (OUT) $BUILD_PREFIX/bin/x86_64-conda_cos6-linux-gnu-c++ -Ofast -D_FILE_OFFSET_BITS=64 -std=c++0x -o tabix++ main.cpp tabix.o -I$SRC_DIR/tabixpp/htslib -L. -L$SRC_DIR/tabixpp/htslib -lhts -lbz2 -lm -lz -llzma -pthread \
00:33:01 BIOCONDA INFO (OUT) 		-lhts -lpthread -lm -lbz2 -llzma -lz
00:33:01 BIOCONDA INFO (OUT) /opt/conda/conda-bld/vcflib_1569889844667/_build_env/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: cannot find -lbz2
00:33:01 BIOCONDA INFO (OUT) /opt/conda/conda-bld/vcflib_1569889844667/_build_env/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: cannot find -lz
00:33:01 BIOCONDA INFO (OUT) /opt/conda/conda-bld/vcflib_1569889844667/_build_env/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: cannot find -llzma
```

**Comment by tseemann** [comment]
@dpryan79 we need `$(CC)` inside the Makefile

**Comment by tseemann** [comment]
```
06:46:02 BIOCONDA INFO (OUT) $BUILD_PREFIX/bin/x86_64-conda_cos6-linux-gnu-c++ -Ofast -D_FILE_OFFSET_BITS=64 -std=c++0x -o tabix++ main.cpp tabix.o -I$SRC_DIR/tabixpp/htslib -L. -L$SRC_DIR/tabixpp/htslib -lhts -lbz2 -lm -lz -llzma -pthread \
06:46:02 BIOCONDA INFO (OUT) 		-lhts -lpthread -lm -lbz2 -llzma -lz
06:46:02 BIOCONDA INFO (OUT) /opt/conda/conda-bld/vcflib_1569912236050/_build_env/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: cannot find -lbz2
06:46:02 BIOCONDA INFO (OUT) /opt/conda/conda-bld/vcflib_1569912236050/_build_env/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: cannot find -lz
06:46:02 BIOCONDA INFO (OUT) /opt/conda/conda-bld/vcflib_1569912236050/_build_env/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: cannot find -llzma
06:46:02 BIOCONDA INFO (OUT) collect2: error: ld returned 1 exit status
```

File: recipes/vcflib/build.sh [hunk 1]
```diff
@@ -26,13 +28,9 @@ if [ "$(uname)" == "Darwin" ]; then
...
 fi
-# tabix missing library https://github.com/ekg/tabixpp/issues/5
-# Uses newline trick for OSX from: http://stackoverflow.com/a/24299845/252589
-#sed -i.bak 's/SUBDIRS=./SUBDIRS=.\'$'\n''LOBJS=tabix.o/' tabixpp/Makefile
-#sed -i.bak 's/-ltabix//' Makefile
-#sed -i.bak 's/make/make -e/' Makefile
 
-make -e
+# -e lets our exported CXX etc override any Makefile settings
+make --environment-overrides -j${CPU_COUNT}
```
Comment (dpryan79) [inline recipes/vcflib/build.sh]: That's a much nicer solution :)

Comment (tseemann) [reply recipes/vcflib/build.sh]: I think I accidently removed `make -e` when I was updating the formula! Whoops :)

Comment (tseemann) [reply recipes/vcflib/build.sh]: Still more to do though. It's a mess.

**Comment by tseemann** [comment]
```
09:16:48 BIOCONDA INFO (OUT) make[3]: Leaving directory `$SRC_DIR/tabixpp/htslib'
09:16:48 BIOCONDA INFO (OUT) $BUILD_PREFIX/bin/x86_64-conda_cos6-linux-gnu-c++ -std=c++0x -Ofast -s -D_FILE_OFFSET_BITS=64 -L. -Ltabixpp/htslib -L$PREFIX/include -lvcflib -lhts -lpthread -lz -lm -llzma -lbz2 -o tabix++ main.cpp tabix.o -I$SRC_DIR/tabixpp/htslib -L. -L$SRC_DIR/tabixpp/htslib -lhts -lbz2 -lm -lz -llzma -pthread \
09:16:48 BIOCONDA INFO (OUT) 		-lhts -lpthread -lm -lbz2 -llzma -lz
09:16:49 BIOCONDA INFO (OUT) /opt/conda/conda-bld/vcflib_1569921279769/_build_env/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: cannot find -lvcflib
09:16:49 BIOCONDA INFO (OUT) collect2: error: ld returned 1 exit status
09:16:49 BIOCONDA INFO (OUT) make[2]: *** [tabix++] Error 1
09:16:49 BIOCONDA INFO (OUT) make[2]: Leaving directory `$SRC_DIR/tabixpp'
```

**Comment by tseemann** [comment]
```
23:25:48 BIOCONDA INFO (OUT) gcc -c filevercmp.c
23:25:48 BIOCONDA INFO (OUT) make[1]: gcc: Command not found
23:25:48 BIOCONDA INFO (OUT) make[1]: *** [filevercmp.o] Error 127
23:25:48 BIOCONDA INFO (OUT) make[1]: Leaving directory `$SRC_DIR/filevercmp'
23:25:48 BIOCONDA INFO (OUT) make: *** [filevercmp/filevercmp.o] Error 2
```

**Comment by tseemann** [comment]
```
23:39:37 BIOCONDA INFO (OUT) $BUILD_PREFIX/bin/x86_64-conda_cos6-linux-gnu-c++ -std=c++0x -Ofast -s -D_FILE_OFFSET_BITS=64 -o tabix++ main.cpp tabix.o -I$SRC_DIR/tabixpp/htslib -L. -L$SRC_DIR/tabixpp/htslib -lhts -lbz2 -lm -lz -llzma -pthread \
23:39:37 BIOCONDA INFO (OUT) 		-lhts -lpthread -lm -lbz2 -llzma -lz
23:39:38 BIOCONDA INFO (OUT) /opt/conda/conda-bld/vcflib_1569973019138/_build_env/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: cannot find -lbz2
23:39:38 BIOCONDA INFO (OUT) /opt/conda/conda-bld/vcflib_1569973019138/_build_env/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: cannot find -lz
23:39:38 BIOCONDA INFO (OUT) /opt/conda/conda-bld/vcflib_1569973019138/_build_env/bin/../lib/gcc/x86_64-conda_cos6-linux-gnu/7.3.0/../../../../x86_64-conda_cos6-linux-gnu/bin/ld: cannot find -llzma
23:39:38 BIOCONDA INFO (OUT) collect2: error: ld returned 1 exit status
23:39:38 BIOCONDA INFO (OUT) make[2]: *** [tabix++] Error 1
23:39:38 BIOCONDA INFO (OUT) make[2]: Leaving directory `$SRC_DIR/tabixpp'
23:39:38 BIOCONDA INFO (OUT) make[1]: *** [all-recur] Error 1
23:39:38 BIOCONDA INFO (OUT) make: *** [tabixpp/tabix.o] Error 2
23:39:38 BIOCONDA INFO (OUT) make: *** Waiting for unfinished jobs....
23:39:38 BIOCONDA INFO (OUT) make[1]: Leaving directory `$SRC_DIR/tabixpp'
23:39:38 BIOCONDA INFO (OUT) $BUILD_PREFIX/bin/x86_64-conda_cos6-linux-gnu-c++ -Ofast -s -D_FILE_OFFSET_BITS=64 -c split.cpp
23:39:39 BIOCONDA INFO (OUT) $BUILD_PREFIX/bin/x86_64-conda_cos6-linux-gnu-c++ -Ofast -s -D_FILE_OFFSET_BITS=64 -c disorder.c
23:39:40 BIOCONDA INFO (OUT) $BUILD_PREFIX/bin/x86_64-conda_cos6-linux-gnu-c++ -Ofast -s -D_FILE_OFFSET_BITS=64 Fasta.o FastaHack.cpp split.o disorder.o -o fastahack
23:39:41 BIOCONDA INFO (OUT) make[1]: Leaving directory `$SRC_DIR/fastahack'
23:39:42 BIOCONDA INFO (OUT) Traceback (most recent call last):
23:39:42 BIOCONDA INFO (OUT)   File "/opt/conda/bin/conda-build", line 11, in <module>
23:39:42 BIOCONDA INFO (OUT)     sys.exit(main())
23:39:42 BIOCONDA INFO (OUT)   File "/opt/conda/lib/python3.7/site-packages/conda_build/cli/main_build.py", line 445, in main
23:39:42 BIOCONDA INFO (OUT)     execute(sys.argv[1:])
23:39:42 BIOCONDA INFO (OUT)   File "/opt/conda/lib/python3.7/site-packages/conda_build/cli/main_build.py", line 436, in execute
23:39:42 BIOCONDA INFO (OUT)     verify=args.verify, variants=args.variants)
23:39:42 BIOCONDA INFO (OUT)   File "/opt/conda/lib/python3.7/site-packages/conda_build/api.py", line 209, in build
23:39:42 BIOCONDA INFO (OUT)     notest=notest, need_source_download=need_source_download, variants=variants)
23:39:42 BIOCONDA INFO (OUT)   File "/opt/conda/lib/python3.7/site-packages/conda_build/build.py", line 2343, in build_tree
23:39:42 BIOCONDA INFO (OUT)     notest=notest,
23:39:42 BIOCONDA INFO (OUT)   File "/opt/conda/lib/python3.7/site-packages/conda_build/build.py", line 1491, in build
23:39:42 BIOCONDA INFO (OUT)     cwd=src_dir, stats=build_stats)
23:39:42 BIOCONDA INFO (OUT)   File "/opt/conda/lib/python3.7/site-packages/conda_build/utils.py", line 397, in check_call_env
```

**Comment by dpryan79** [comment]
It compiles on Linux now, not sure what the issue is on OSX.

**Comment by cbrueffer** [comment]
Outdated; vcflib is up to date at 1.0.3.

**Pull request closed without merging**